### DISCUSSION TITLE
Using verify with custom verified file name

### DISCUSSION THREAD

**Discussion by lahma** [VerifyTests/Verify#545] [Q&A] [answered]
I'm investigating how Verify could be used testing [esprima-dotnet](https://github.com/sebastienros/esprima-dotnet). This means that we have bunch of JS files and their expected produced AST as JSON file. The common pattern among parsers / test suites is to have `myfilename.js` and corresponding `myfilename.tree.json`, but this is not set in stone (there might another naming too). But generally there are thousands of files available following this file-pair-naming for people wanting to implement parsers and verifying correct outcome.

[Here's an example directory of JS input and expected AST JSON output](https://github.com/sebastienros/esprima-dotnet/tree/main/test/Esprima.Tests/Fixtures/es2018/dynamic-import).

Looking at these lines:

https://github.com/VerifyTests/Verify/blob/350bd3f730ad47ad8a3e2c8104703993f5de3326/src/Verify/FilePair.cs#L23-L24

It seems that the expected file name handling is quite set in stone. Are the any recommendations or ideas how to handle such custom workflow where `input file -> parser -> in-memory-output` is handled where one would like to use existing file naming without tightly coupling to Verify (basically allowing to use those expected files out in the wild, might be a git subtree too so cannot rename files).

I guess I'm asking for a feature to globally (or even per verify run) to provide a custom file name formatter, unless this can be achieved already otherwise?

**Comment by SimonCropp** [comment, marked as answer]
yes. currently the `.verified.` and `.verified.` are hard coded.

> basically allowing to use those expected files out in the wild, might be a git subtree too so cannot rename files

I am not sure Verify provided much value in this case. Much of the value proposition is in the ability to automate the acceptance of changed files. In your case you never want to accept those changes. I suspect what you want is the diffing of file functionality. you can get this by using the lower level DiffEngine package https://github.com/VerifyTests/DiffEngine#launching-a-tool

**Reply by lahma** [reply]
Thanks for the valuable info. I'll be investigating the DiffEngine, thanks for pointing to it!
